plugins/modules/net_tools/nmcli.py

```diff
@@ -183,6 +183,18 @@
         type: str
         choices: [ignore, auto, dhcp, link-local, manual, shared, disabled]
         version_added: 2.2.0
+    ip_privacy6:
+        description:
+            - If enabled, it makes the kernel generate a temporary IPv6 address in addition to the public one.
+        type: str
+        choices: [disabled, prefer-public-addr, prefer-temp-addr, unknown]
+        version_added: 4.2.0
+    addr_gen_mode6:
+        description:
+            - Configure method for creating the address for use with IPv6 Stateless Address Autoconfiguration.
+        type: str
+        choices: [eui64, stable-privacy]
```

Comment:
Were you able to actually test this setting in a real environment? I only ask because this defaults to `stable-privacy` and `nmcli` tends to hide defaults until the setting is explicitly set (which is why we tend to have issues with defaults affecting changed status)

Reply:
First of all, i was able to test setting in a real environment. Here is what i did:
i ran this playbook:
```
---
- name: checking stable-privacy and prefer-public-addr
  hosts: [redhat]
  become: true
  tasks:
    - name: clear dummy_con1
      nmcli:
        conn_name: dummy_con1
        ifname: dummy0
        state: absent
    
    - name: Adding new dummy connection with ip6 address
      nmcli:
        conn_name: dummy_con1
        ifname: dummy0
        ip6: 2001:db8::cafe/128
        gw6: 2001:db8::cafa
        dns6: 2001:4860:4860::8888
        state: present
        type: dummy

    - name: Adding stable-privacy and prefer-public-addr
      nmcli:
        conn_name: dummy_con1
        ifname: dummy0
        ip6: 2001:db8::cafe/128
        gw6: 2001:db8::cafa
        dns6: 2001:4860:4860::8888
        ip_privacy6: prefer-public-addr
        addr_gen_mode6: stable-privacy
        state: present
        type: dummy

    - name: repeating last task
      nmcli:
        conn_name: dummy_con1
        ifname: dummy0
        ip6: 2001:db8::cafe/128
        gw6: 2001:db8::cafa
        dns6: 2001:4860:4860::8888
        ip_privacy6: prefer-public-addr
        addr_gen_mode6: stable-privacy
        state: present
        type: dummy
```
and got this results:
```
ansible-playbook bad_playbook.yaml --diff

PLAY [checking stable-privacy and prefer-public-addr] *******************************************************************************************************

TASK [Gathering Facts] **************************************************************************************************************************************
ok: [redhat]

TASK [clear dummy_con1] *************************************************************************************************************************************
changed: [redhat]

TASK [Adding new dummy connection with ip6 address] *********************************************************************************************************
changed: [redhat]

TASK [Adding stable-privacy and prefer-public-addr] *********************************************************************************************************
--- before
+++ after
@@ -15,6 +15,6 @@
     "ipv6.gateway": "2001:db8::cafa",
     "ipv6.ignore-auto-dns": "no",
     "ipv6.ignore-auto-routes": "no",
-    "ipv6.ip6-privacy": "-1",
+    "ipv6.ip6-privacy": "1 (enabled, prefer public IP)",
     "ipv6.method": "manual"
 }

changed: [redhat]

TASK [repeating last task] **********************************************************************************************************************************
ok: [redhat]

PLAY RECAP **************************************************************************************************************************************************
redhat                     : ok=5    changed=3    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0
```
So it works as you would expect.
I have also tested all other options, so if you need more real environment tests i can send it.

Second of all:
Suggested change make a lot of tests failed because of situation when ip_privacy6 is None.
I think we can also fix this by deleting ```ipv6.ip6-privacy``` option from here:
```
# IP address options.
        if self.ip_conn_type and not self.master:
            options.update({
                'ipv4.addresses': self.ip4,
                'ipv4.dhcp-client-id': self.dhcp_client_id,
                'ipv4.dns': self.dns4,
                'ipv4.dns-search': self.dns4_search,
                'ipv4.ignore-auto-dns': self.dns4_ignore_auto,
                'ipv4.gateway': self.gw4,
                'ipv4.ignore-auto-routes': self.gw4_ignore_auto,
                'ipv4.routes': self.routes4,
                'ipv4.route-metric': self.route_metric4,
                'ipv4.routing-rules': self.routing_rules4,
                'ipv4.never-default': self.never_default4,
                'ipv4.method': self.ipv4_method,
                'ipv4.may-fail': self.may_fail4,
                'ipv6.addresses': self.ip6,
                'ipv6.dns': self.dns6,
                'ipv6.dns-search': self.dns6_search,
                'ipv6.ignore-auto-dns': self.dns6_ignore_auto,
                'ipv6.gateway': self.gw6,
                'ipv6.ignore-auto-routes': self.gw6_ignore_auto,
                'ipv6.method': self.ipv6_method,
                'ipv6.ip6-privacy': self.ip_privacy6,
                'ipv6.addr-gen-mode': self.addr_gen_mode6
            })
```
 but doesn't look like the right way to do it.

And the last one:
If you use ```assert``` sanity test would fail with: ```raise AssertionError instead of: assert```
so i have changed it, as sanity test suggest

Reply:
- As long as the basic features work no more testing is needed from me (any bugs can be addressed when/if they appear). 
- The guard for `None` you added is good.
- I guess lint doesn't want code which can be disabled so that makes sense. If you want you can take out the `AssertionError` bit and just let it throw a `KeyError` if there's a bad update at some point (as long as it doesn't allow bad keys to go unnoticed I'm good).

Reply:
I'm ok with ```AssertionError```.
Strangely there only one test ran on my last commit. It used to be at about 146 tests.

Reply:
Uh, AZP might catch-up at some point, but we can trigger it before merge if it doesn't.